app/sessionStore.js

```diff
@@ -847,6 +847,16 @@ module.exports.runPreMigrations = (data) => {
       if (data.cache) {
         delete data.cache.bookmarkLocation
       }
+
+      // pinned top sites were stored in the wrong position in 0.19.x
+      // allowing duplicated items. See #12941
+      // in this case eliminate pinned items so they can be properly
+      // populated in their own indexes
+      if (data.about.newtab.pinnedTopSites) {
+        // Empty array is currently set to include default pinned sites
+        // which we avoid given the user already have a profile
+        data.about.newtab.pinnedTopSites = [null]
```

Comment:
How come we're setting this to an array with one `null` item inside and not an empty array with `[]` @cezaraugusto ?

Reply:
see https://github.com/brave/browser-laptop/blob/master/app/common/state/aboutNewTabState.js#L41 an empty array would populate top sites with default pinned site, which should only happen on an empty profile

Reply:
could be done with something like `Array.from({length: 18}, _ => null)` as pinned top sites are set as 18 null spaces but conversion is covered already in both newTab and newTabState